lib/configurators/aws/aws_test.go

```diff
@@ -657,7 +657,49 @@ func TestAWSIAMDocuments(t *testing.T) {
 				{
 					Effect:    awslib.EffectAllow,
 					Resources: awslib.SliceOrString{"*"},
-					Actions:   awslib.SliceOrString{"redshift-serverless:GetEndpointAccess", "redshift-serverless:GetWorkgroup", "redshift-serverless:ListWorkgroups", "redshift-serverless:ListEndpointAccess", "redshift-serverless:ListTagsForResource"},
+					Actions:   awslib.SliceOrString{"redshift-serverless:GetEndpointAccess", "redshift-serverless:GetWorkgroup", "redshift-serverless:ListWorkgroups", "redshift-serverless:ListEndpointAccess", "redshift-serverless:ListTagsForResource", "sts:AssumeRole"},
+				},
+			},
+		},
+		"AWS Keyspaces static databases": {
+			target: roleTarget,
+			fileConfig: &config.FileConfig{
+				Databases: config.Databases{
+					Databases: []*config.Database{{
+						Name:     "keyspaces-1",
+						Protocol: "cassandra",
+						AWS: config.DatabaseAWS{
+							AccountID: "123456789012",
+						},
+					}},
+				},
+			},
+			boundaryStatements: []*awslib.Statement{
+				{
+					Effect:    awslib.EffectAllow,
+					Resources: awslib.SliceOrString{"*"},
+					Actions:   awslib.SliceOrString{"sts:AssumeRole"},
+				},
+			},
+		},
+		"DynamoDB static databases": {
+			target: roleTarget,
+			fileConfig: &config.FileConfig{
+				Databases: config.Databases{
+					Databases: []*config.Database{{
+						Name:     "dynamodb-1",
+						Protocol: "dynamodb",
+						AWS: config.DatabaseAWS{
+							Region: "us-west-1",
+						},
+					}},
+				},
+			},
+			boundaryStatements: []*awslib.Statement{
+				{
+					Effect:    awslib.EffectAllow,
+					Resources: awslib.SliceOrString{"*"},
+					Actions:   awslib.SliceOrString{"sts:AssumeRole"},
```

Comment:
Adding `"*"` is simple but I am not sure users will be intimidated by it for `sts:AssumeRole`. While since this is the boundary policy, it doesn't have any effect on itself. But if we feel users may have a problem with it, we can also have a condition, e.g.
```
        {
            "Effect": "Allow",
            "Action": [
                "sts:AssumeRole"
            ],
            "Resource": "*",
            "Condition": {
                "StringEquals": {
                    "aws:ResourceTag/TeleportAssumeRole": "true"
                }
            }
        }
```
Obviously, doing it this way means all "access" roles need to have tag in order to be assumed.

Reply:
Interesting, I thought the trust policy on a role can alone determine if another entity can assume the role

Reply:
> Interesting, I thought the trust policy on a role can alone determine if another entity can assume the role

That's what I thought too but seems the trust relationship does NOT affect the permission boundary so it has to be added explicitly.

Reply:
I had completely forgotten about this change and ran into this problem while testing out assume role chaining using a role I had previously bootstrapped with Teleport, so it had the old boundary policy. I thought I was going to have to update our boundary policy creation 😅. Thanks again for this fix!